packages/react/test/specs/components/Dropdown/Dropdown-test.tsx

```diff
@@ -34,22 +34,207 @@ describe('Dropdown', () => {
+
+      expect(onOpenChange).toBeCalledTimes(2)
+      expect(onOpenChange).toHaveBeenLastCalledWith(
+        null,
+        expect.objectContaining({
+          open: false,
+        }),
+      )
+    })
+
+    it('is false when closed by hitting Escape in search input', () => {
+      const wrapper = mountWithProvider(
+        <Dropdown onOpenChange={onOpenChange} search items={items} />,
+      )
+
+      wrapper
+        .find(`input.${DropdownSearchInput.slotClassNames.input}`)
+        .simulate('click')
+        .simulate('change', { target: { value: 'test' } })
+        .simulate('keydown', { keyCode: keyboardKey.Escape, key: 'Escape' })
+
+      expect(onOpenChange).toBeCalledTimes(2)
+      expect(onOpenChange).toHaveBeenLastCalledWith(
+        null,
+        expect.objectContaining({
+          open: false,
+        }),
+      )
+    })
+
+    it('is false when closed by hitting Escape in items list', () => {
+      const wrapper = mountWithProvider(
+        <Dropdown onOpenChange={onOpenChange} multiple items={items} />,
+      )
+
+      wrapper
+        .find(`button.${Dropdown.slotClassNames.triggerButton}`)
+        .simulate('click')
+        .simulate('keydown', { keyCode: keyboardKey.Escape, key: 'Escape' })
+
+      expect(onOpenChange).toBeCalledTimes(2)
+      expect(onOpenChange).toHaveBeenLastCalledWith(
+        null,
+        expect.objectContaining({
+          open: false,
+        }),
+      )
+    })
+
+    it('is false when an item has been selected', () => {
+      const wrapper = mountWithProvider(<Dropdown onOpenChange={onOpenChange} items={items} />)
+
+      wrapper.find(`button.${Dropdown.slotClassNames.triggerButton}`).simulate('click')
+      wrapper
+        .find(`li.${Dropdown.slotClassNames.item}`)
+        .at(0)
+        .simulate('click')
+
+      expect(onOpenChange).toBeCalledTimes(2)
+      expect(onOpenChange).toHaveBeenLastCalledWith(
+        null,
+        expect.objectContaining({
+          open: false,
+        }),
+      )
+    })
+  })
+
   describe('highlightedIndex', () => {
     const onOpenChange = jest.fn()
 
     afterEach(() => {
       onOpenChange.mockReset()
     })
 
+    it('is null when opened by click', () => {
+      const wrapper = mountWithProvider(<Dropdown onOpenChange={onOpenChange} items={items} />)
+
+      wrapper.find(`button.${Dropdown.slotClassNames.triggerButton}`).simulate('click')
+
+      expect(onOpenChange).toBeCalledTimes(1)
+      expect(onOpenChange).toHaveBeenLastCalledWith(
+        null,
+        expect.objectContaining({
+          highlightedIndex: null,
+          open: true,
+        }),
+      )
+    })
+
+    it.skip('is null when opened by Enter', () => {
+      const wrapper = mountWithProvider(<Dropdown onOpenChange={onOpenChange} items={items} />)
+
+      wrapper
+        .find(`button.${Dropdown.slotClassNames.triggerButton}`)
+        .simulate('focus')
+        .simulate('keydown', { keyCode: keyboardKey.Enter, key: 'Enter' })
```

Comment:
I was not able to simulate this even in browser, consider to remove these assertions. Should notice that they assert for existing browser behavior...